### ISSUE TITLE
Expecting to find application/json in Content-Type header

### ISSUE THREAD

**Issue by mariancelo** [redbo/cloudfuse#75]
Cloudfuse stopped working few days ago.

I am using "openstack type" storage and when I try to mount a storage I got this error message:

_Entity: line 1: parser error : Document is empty_
_{"error": {"message": "Expecting to find application/json in Content-Type header_
_^_
_Failed to authenticate._

I haven't touched my config file (_~/.cloudfuse_)

It looks like something in "which headers openstack is accepting has changed", as I looked at the code cloudfuse is using application/xml content-type

Anyone has similar problem?

**Comment by versgui** [comment]
Exactly the same problem here, I don't know what happened and I didn't find any log...

**Comment by mariancelo** [comment]
I am using runabove storage. Maybe it has something to do with upgrade on storage system they did on August the 12th. (bottom of the page: https://status.runabove.com/)

Just to be sure that cloudfuse stopped working on runabove only. versgui: are you using the same storage?

thnx

**Comment by versgui** [comment]
Yes, I'm using Runabove too. The update of their authentication system may explain the problem.

**Comment by mariancelo** [comment]
ok, thanks,  I have already written a ticket(2 days ago) to their support, but I didn't mention that it could be connected with their security update. I will update the ticket and will let you know, when they answer it.

**Comment by versgui** [comment]
Thank you very much!

**Comment by ndriu** [comment]
Hello,

did you get any answer from support? I have the same problem...

Andrzej

**Comment by mariancelo** [comment]
This is what support has answered:

"Hello, I checked with our administrators and the maintenance is now over. If you still have an authentification error, I suggest you to check if you can log in in your Horizon Dashboard. If it works, I suggest you to check directly with the cloudfuse community since everything is okay on our side."

**Comment by strakadavid** [comment]
Hello, I have the same problem. Can you make fix? It looks like something changed.

**Comment by MedericB** [comment]
Hi, did you find any solution ?
I have the same problem with https://auth.cloud.ovh.net/v2.0 (cloud like Runabove by OVH).

**Comment by digitalmio** [comment]
Anyone managed to fix this? I have the same problem using OVH ObjectStorage

**Comment by andrewgaul** [comment]
You should consider using [s3fs](https://www.eclipse.org/jetty/documentation/current/setting-form-size.html) or [goofys](https://github.com/kahing/goofys) with [S3Proxy](https://github.com/andrewgaul/s3proxy) using the Swift backend as a workaround until this issue can be fixed.

**Comment by Jaimy85** [comment]
It seems OVH only accepts json format, cloudfuse only works with xml.
snprintf(postdata, sizeof(postdata), "{\"auth\": {\"tenantName\": \"%s\", \"passwordCredentials\":" 
        "{\"username\": \"%s\", \"password\": \"%s\" } } }"
        , reconnect_args.tenant, reconnect_args.username, reconnect_args.password);
I use this it will give you the auth, however more code is needed to handle the json response.

**Comment by cortopy** [comment]
Are there any updates on this? I'm getting the same problem two months after

**Comment by nikuzz** [comment]
No new updates on this for OVH ObjectStorage?
@Jaimy85 where I can change the code like you suggest?
For the moment I try to use [rclone](http://rclone.org/), but it's not the same because you need to have a local copy of the data, or to manage a "copy and delete" for a "backup use"

**Comment by dakser** [comment]
Hi, excuse me, do you have founded any solution to this? I have this problem with OVH Object Storage and their support doesn't help. Thanks in advance.

**Comment by nikuzz** [comment]
Today I'll try with hubicfuse... maybe it works with the recular Swift Object Storage.

**Comment by nikuzz** [comment]
@dakser I confirm that with hibicfuse you cannot reach the goal, OVH Object storage has different API then hubic :(
I finally try [s3ql](https://github.com/s3ql/s3ql) and seems working well.
Here a guide from OVH:
[https://www.ovh.com/fr/g1908.monter_un_conteneur_dobjet_avec_s3ql](https://www.ovh.com/fr/g1908.monter_un_conteneur_dobjet_avec_s3ql)

It works also with rclone, but in this case you need to manage the sync and keep a local copy, simple it is not a mount but only a clone. Bye

**Comment by dakser** [comment]
Thanks a lot nikuzz, yes, it looks like S3QL works without any problem, but
it creates a container inside the Object Storage and that not suits my
needs for managing the files inside the bucket. I can only hope to be
solved to use OVH Object Storage with the simplicity of mounting it as a
local drive. Thanks for your help!

2016-03-09 8:11 GMT-06:00 nikuzz <notifications@github.com>:

**Comment by dakser** [comment]
Hi, I have founded a new proyect that's from OVH to mount Swift, right now
I'm testint it. https://github.com/ovh/svfs

2016-03-09 18:17 GMT-06:00 Dax2424 <aguilatrix@gmail.com>:

**Comment by cortopy** [comment]
@dakser thanks for mentioning it! Sounds like the right solution

**Comment by kdahmani** [comment]
@dakser thanks for your work, I am having problems connecting using this command:
sudo mount -t svfs -o username=99999999999xxx,password=xxxxxxxxxxx,tenant=xxxxxxxxxxx,region=bhs1 pcs /storage

I would like to know where we get the following from in the OVH control panel:

username
password
tenant

I keep on getting Authorization failed if I use swift I can see my storage with this command

swift --os-auth-token xxxxxxa4ec6478eaf580f629xxxxxx --os-storage-url https://storage.bhs1.cloud.ovh.net/v1/AUTH_xxxxxxx0c0b445959c6d1df1xxxxxxxx list
